internal/speakerlist/speakerlist.go

```diff
@@ -231,7 +234,49 @@ func (sl *SpeakerList) members() map[string]struct{} {
 
 // SetSpeakersK8S update the list of speakers from the K8S API point of view
 func (sl *SpeakerList) SetSpeakersK8S(eps k8s.EpsOrSlices) {
-	// to be implemented
+	newmap := map[string]string{}
+
+	switch eps.Type {
+	case k8s.Eps:
+		for _, subset := range eps.EpVal.Subsets {
+			for _, ep := range subset.Addresses {
+				if ep.NodeName == nil {
+					continue
+				}
+				newmap[*ep.NodeName] = ep.IP
+			}
+		}
+	case k8s.Slices:
+		for _, slice := range eps.SlicesVal {
+			for _, ep := range slice.Endpoints {
+				if !k8s.IsConditionReady(ep.Conditions) {
+					continue
+				}
+				nodeName := ep.Topology["kubernetes.io/hostname"]
+				if nodeName == "" {
+					continue
+				}
+				newmap[nodeName] = ep.Addresses[0]
+			}
+		}
+	}
+
+	sl.Lock()
+	oldmap := sl.k8sSpeakers
+	sl.Unlock()
+	if reflect.DeepEqual(oldmap, newmap) {
+		return
+	}
+	sl.l.Log("op", "setSpeakersK8S", "msg", "New K8S Speakers list set", "old-list", oldmap, "new-list", newmap)
```

Comment:
In a big cluster, both of these lists could be hundreds of endpoints, right? Will that cause a problem? I don't remember where these logs end up (e.g. do they get thrown into k8s events?)

Reply:
Logs will only end up in normal logs, I'm now just showing the added and removed nodes